src/test/16_IntegrationManager.t.sol

```diff
@@ -0,0 +1,147 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity ^0.8.13;
+
+import "../../lib/forge-std/src/Test.sol";
+import "../Integrations/IntegrationManager.sol";
+import "../mockups/AdapterMock.sol";
+import "../Vault.sol";
+
+abstract contract IntegrationManagerTest is Test {
+    using stdStorage for StdStorage;
+
+    Vault vault;
+    IntegrationManager im;
+    AdapterMock adapter;
+
+    address deployer = address(1);
+    address vaultOwner = address(2);
+    address mainRegistry = address(3);
+
+    //Before
+    constructor() {
+        
+    }
+
+    //Before Each
+    function setUp() public virtual {
+        vm.startPrank(deployer);
+        vault = new Vault();
+        im = new IntegrationManager(address(mainRegistry));
+        adapter = new AdapterMock(address(im));
+        vm.stopPrank();
+
+        // Cheat owner
+        uint256 slot = stdstore.target(address(vault)).sig(vault.owner.selector).find();
```

Comment:
https://github.com/arcadia-finance/arcadia-vaults/blob/a302005bfdd6637c2aeb194b03d4496f0296de23/src/test/unit/RiskModule.t.sol#L30
I used .with_key(...).checked_write(...) to make this one liner